pkg/server/api.go

```diff
@@ -80,6 +80,10 @@ func NewServerAPIHandler(s Server) *APIHandler {
 // ServeHTTP handles the requests for the machine config server
 // API handler.
 func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
```

Comment:
@wking what about `HEAD`. Somebody might send head for headers before sending `GET` ?

but i'm punting on that before to follow up later.

Reply:
> @wking what about `HEAD`.

I've added support for `HEAD` and touched up our header handling with 2dc59a3 -> 22d0c45.